scylla/src/frame/value_tests.rs

```diff
@@ -723,3 +724,156 @@ async fn test_cqlvalue_udt() {
 
     assert_eq!(received_udt_cql_value, &udt_cql_value);
 }
+
+#[tokio::test]
+async fn test_cqlvalue_duration() {
+    let uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session: Session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+    let ks = crate::transport::session_test::unique_name();
+    session
+        .query(
+            format!(
+                "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = \
+            {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}",
+                ks
+            ),
+            &[],
+        )
+        .await
+        .unwrap();
+    session.use_keyspace(&ks, false).await.unwrap();
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS cqlvalue_duration_test (k int, my duration, primary key (k))",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let duration_cql_value = CqlValue::Duration(CqlDuration {
+        months: 6,
+        days: 9,
+        nanoseconds: 21372137,
+    });
+
+    session
+        .query(
+            "INSERT INTO cqlvalue_duration_test (k, my) VALUES (5, ?)",
+            (&duration_cql_value,),
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            "INSERT INTO cqlvalue_duration_test (k, my) VALUES (6, 89h4m48s)",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            "INSERT INTO cqlvalue_duration_test (k, my) VALUES (7, PT89H8M53S)",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            "INSERT INTO cqlvalue_duration_test (k, my) VALUES (8, P0000-00-00T89:09:09)",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let rows = session
+        .query(
+            "SELECT my FROM cqlvalue_duration_test WHERE k = ?",
+            (CqlValue::Int(5),),
+        )
+        .await
+        .unwrap()
+        .rows
+        .unwrap();
+
+    assert_eq!(rows.len(), 1);
+    assert_eq!(rows[0].columns.len(), 1);
+
+    let received_duration_cql_value = rows[0].columns[0].as_ref().unwrap();
+
+    assert_eq!(received_duration_cql_value, &duration_cql_value);
+
+    let rows = session
```

Comment:
Instead of fetching every row in a separate select, you can run a single select and fetch all rows from the table. If you want to have them in a sorted order, then you can use a single partition and use different clustering keys.